internal/bminventory/inventory.go

```diff
@@ -1965,10 +1966,12 @@ func (b *bareMetalInventory) updateClusterData(_ context.Context, cluster *commo
 			return common.NewApiError(http.StatusBadRequest, errors.Errorf(msg))
 		}
 	}
-
-	dbReply := db.Model(&common.Cluster{}).Where("id = ?", cluster.ID.String()).Updates(updates)
-	if dbReply.Error != nil {
-		return common.NewApiError(http.StatusInternalServerError, errors.Wrapf(err, "failed to update cluster: %s", params.ClusterID))
+	if len(updates) > 0 {
+		updates["trigger_monitor_timestamp"] = time.Now()
+		dbReply := db.Model(&common.Cluster{}).Where("id = ?", cluster.ID.String()).Updates(updates)
```

Comment:
why not to have common for some cluster_api function for updateCluster in db?

Reply:
Not sure I agree.  Anyway, if it should be done - it should be part of a bigger change, not just these changes